examples/routing/delegated-routing-client/main.go

```diff
@@ -0,0 +1,142 @@
+	if err := run(os.Stdout, *gatewayUrlPtr, *cidPtr, *pidPtr, *namePtr, *timeoutPtr); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run(w io.Writer, gatewayURL, cidStr, pidStr, nameStr string, timeoutSeconds int) error {
+	// Creates a new Delegated Routing V1 client.
+	client, err := client.New(gatewayURL)
+	if err != nil {
+		return err
+	}
+
+	timeout := time.Duration(timeoutSeconds) * time.Second
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	if cidStr != "" {
+		return findProviders(w, ctx, client, cidStr)
+	} else if pidStr != "" {
+		return findPeers(w, ctx, client, pidStr)
+	} else if nameStr != "" {
+		return findIPNS(w, ctx, client, nameStr)
+	} else {
+		return errors.New("cid or peer must be provided")
+	}
+}
+
+func findProviders(w io.Writer, ctx context.Context, client *client.Client, cidStr string) error {
+	// Parses the given CID to lookup the providers for.
+	contentCid, err := cid.Parse(cidStr)
+	if err != nil {
+		return err
+	}
+
+	// Ask for providers providing the given content CID.
+	recordsIter, err := client.FindProviders(ctx, contentCid)
+	if err != nil {
+		return err
+	}
+	defer recordsIter.Close()
+	return printIter(w, recordsIter)
+}
+
+func findPeers(w io.Writer, ctx context.Context, client *client.Client, pidStr string) error {
+	// Parses the given Peer ID to lookup the information for.
+	pid, err := peer.Decode(pidStr)
+	if err != nil {
+		return err
+	}
+
+	// Ask for information about the peer with the given peer ID.
+	recordsIter, err := client.FindPeers(ctx, pid)
+	if err != nil {
+		return err
+	}
+	defer recordsIter.Close()
+	return printIter(w, recordsIter)
+}
+
+func printIter(w io.Writer, iter iter.ResultIter[types.Record]) error {
+	// The response is streamed. Alternatively, you could use [iter.ReadAll]
+	// to fetch all the results all at once, instead of iterating as they are
+	// streamed.
+	for iter.Next() {
+		res := iter.Val()
+
+		// Check for error, but do not complain if we exceeded the timeout. We are
+		// expecting that to happen: we explicitly defined a timeout.
+		if res.Err != nil {
+			if !errors.Is(res.Err, context.DeadlineExceeded) {
+				return res.Err
+			}
+
+			return nil
+		}
+
+		switch res.Val.GetSchema() {
+		case types.SchemaPeer:
+			record := res.Val.(*types.PeerRecord)
+			fmt.Fprintln(w, record.ID)
+			fmt.Fprintln(w, "\tProtocols:", record.Protocols)
+			fmt.Fprintln(w, "\tAddresses:", record.Addrs)
+		default:
+			// You may not want to fail here, it's up to you. You can just handle
+			// the schemas you want, or that you know, but not fail.
+			log.Printf("unrecognized schema: %s", res.Val.GetSchema())
+		}
+	}
+
+	return nil
+}
+
+func findIPNS(w io.Writer, ctx context.Context, client *client.Client, nameStr string) error {
+	// Parses the given name string to get a record for.
+	name, err := ipns.NameFromString(nameStr)
+	if err != nil {
+		return err
+	}
+
+	// Ask for a record. [client.GetIPNS] validates the record for us.
```

Comment:
We should not teach people to trust remote validation of IPNS. 
Could we add local signature check and at elast print `Signature: VALID|INVALID` below `Value` ?

Reply:
@lidel hmm, it's not remote validation though. We can add a validation __again__, but it will never be invalid because the client code already validates it.

Reply:
@lidel hmm, it is not remote validation though. It is very local validation inside the `GetIPNS` function:

https://github.com/ipfs/boxo/blob/5ae64f335dd4ccedcb5b63108ccc8fe8e4408843/routing/http/client/client.go#L408-L444

I updated the comments to make it 100% clear that the record is validated by Boxo client. Also added that when using other clients or retrieving through other methods, records should always be validated.

I am going ahead and merge, but if you feel like we should really add a _second validation_ that will always succeed for some clarity, I can do it in a following PR.